### ISSUE TITLE
<<< (SinOsc s => Pan2 p) >>>; causes miniAudicle to crash

### ISSUE THREAD

**Issue by nshaheed** [ccrma/chuck#419]
```
<<< (SinOsc s => Pan2 p) >>>;
```

This causes miniaudicle to crash, and cli chuck to spit this out:

```
chuck: chuck_scan.cpp:2370: long unsigned int type_engine_scan2_exp_decl_create(Chuck_Env*, a_Exp_Decl): Assertion `type != NULL' failed.
Aborted
```

**Comment by gewang** [comment]
```
// Assertion failed: (type != NULL),
// function type_engine_scan2_exp_decl_create,
// file chuck_scan.cpp, line 2306.

class Foo
{
    1 => float num;
    fun void Foo( int x ) { <<< "constructor!", x >>>; x => num; }
}

<<< (Foo foo(5)).num >>>;
```

**Comment by gewang** [comment]
issue due to missing scanning phase for exp_primary; added content for scan1_exp_primary() and scan2_exp_primary(); addressed in 1edeec61302f1ee2c69ea7239d0b66a74ca653fb
